src/outbox/outboxReducer.js

```diff
@@ -17,7 +17,15 @@ const initialState = NULL_ARRAY;
 const messageSendStart = (state, action) => {
   const message = state.find(item => item.timestamp === action.outbox.timestamp);
   if (message) {
-    return state;
+    // If we find there is already a message in the outbox with the same
```

Comment:
Looks like there's a test in `outboxReducer-test.js` that should be changed to test the new behavior. 🙂

Reply:
Also, an extra-principled option would be to add an `outboxActions-test.js`, like we have `fetchActions-test.js`, on the action side of Redux.

Mainly, `fetchActions-test.js` tests the async `thunk` action creators to make sure they dispatch the appropriate plain object actions when they're supposed to. Note that `redux-mock-store` describes in its README that it doesn't do anything with the reducers; there's not currently a clear way to test the actions and the reducers together.

Could just start with something small, so that there's at least something to add to, for the next person making changes to the outbox actions.

Reply:
> Looks like there's a test in `outboxReducer-test.js` that should be changed to test the new behavior. 🙂

This is no longer true, with the implementation Greg suggests at https://github.com/zulip/zulip-mobile/pull/4129#discussion_r463413900.

Reply:
> Also, an extra-principled option would be to add an `outboxActions-test.js`, like we have `fetchActions-test.js`, on the action side of Redux.

I took a stab at doing this, but I think it would be best to do this along with #3881, so that we don't end up doing work that we'll just discard.